deployment/common/changeset/transfer_link_token.go

```diff
@@ -0,0 +1,71 @@
+package changeset
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+var _ deployment.ChangeSet[TransferLinkTokenBatchConfig] = TransferLinkTokenBatch
+
+type TokenTransferer interface {
+	Transfer(opts *bind.TransactOpts, to common.Address, amount *big.Int) (*gethtypes.Transaction, error)
+}
+
+type TransferLinkTokenConfig struct {
+	To         common.Address
+	Amount     *big.Int
+	Transferer TokenTransferer
+}
+
+type TransferLinkTokenBatchConfig struct {
+	Transfers map[uint64]TransferLinkTokenConfig
+}
+
+func (cfg TransferLinkTokenBatchConfig) Validate() error {
+	for k, c := range cfg.Transfers {
+		if err := deployment.IsValidChainSelector(k); err != nil {
+			return err
+		}
+		if c.To == (common.Address{}) {
+			return errors.New("to address must be set")
+		}
+		if c.Amount == nil || c.Amount.Sign() == -1 {
+			return errors.New("amount must be set and positive")
+		}
+		if c.Transferer == nil {
+			return errors.New("transferer must be set")
+		}
+	}
+
+	return nil
+}
+
+// TransferLinkTokenBatch transfers link token to the to address from the transferer in a batch.
+func TransferLinkTokenBatch(e deployment.Environment, config TransferLinkTokenBatchConfig) (deployment.ChangesetOutput, error) {
```

Comment:
We'll need the changeset that returns the transfer MCMS proposal, you can include it in this PR as well